cloudify/amqp_client.py

```diff
@@ -413,14 +414,20 @@ def _process_message(self, channel, properties, full_task, delivery_tag):
             if result is NO_RESPONSE:
                 self.delete_queue(properties.reply_to)
             else:
+                if result is STOP_AGENT:
+                    body = json.dumps({'ok': True})
+                else:
+                    body = json.dumps(result)
                 self._connection.publish({
                     'exchange': self.exchange,
                     'routing_key': properties.reply_to,
                     'properties': pika.BasicProperties(
                         correlation_id=properties.correlation_id),
-                    'body': json.dumps(result)
+                    'body': body
                 })
-
+        if result is STOP_AGENT:
+            # the operation asked us to exit, so drop everything and exit
+            os._exit(0)
```

Comment:
when I say "drop everything", I mean it. Hence the very forceful `_exit`.
The previous approach was to `kill -9` the daemon from outside so this is just a bit nicer.

(regular `sys.exit` would wait for all the threads to finish, which doesn't really ever happen)